tests/util/translator/translator.spec.tsx

```diff
@@ -0,0 +1,311 @@
+            uiType: EElementType.EMAIL,
+        }),
+    }
+);
+
+const MOCK_EMAIL_SCHEMA_WITH_VALIDATION = generateMockSchema(
+    {},
+    {
+        ...generateMockElementSchema({
+            id: "mockId1",
+            label: "Email address",
+            uiType: EElementType.EMAIL,
+            validation: [
+                {
+                    required: true,
+                    errorMessage: "Email address is required",
+                },
+                {
+                    matches: "/^[a-zA-Z0-9._%+-]+@(gmail\\.com)$/",
+                    errorMessage:
+                        "Enter a email that has a '@gmail.com' domain",
+                },
+            ],
+        }),
+    }
+);
+
+const MOCK_EMAIL_SCHEMA_WITH_CONDITIONAL_RENDERING = generateMockSchema(
+    {},
+    {
+        ...generateMockElementSchema({
+            id: "mock123",
+            label: "Email address",
+            uiType: EElementType.EMAIL,
+            validation: [
+                {
+                    required: true,
+                    errorMessage: "Email address is required",
+                },
+            ],
+            showIf: [
+                {
+                    mock456: [
+                        { filled: true },
+                        {
+                            equals: "hello",
+                        },
+                    ],
+                },
+            ],
+        }),
+        ...generateMockElementSchema({
+            id: "mock456",
+            label: "Short text",
+            uiType: EElementType.TEXT,
+            validation: [
+                {
+                    required: true,
+                    errorMessage: "Input is required",
+                },
+            ],
+        })
+    }
+);
+
+const MOCK_EMAIL_SCHEMA_WITH_MYINFO_PREFILL = generateMockSchema(
+    {
+        mock123: [
+            {
+                prefillMode: "Myinfo",
+                path: "testpath",
+            },
+        ],
+    },
+    {
+        ...generateMockElementSchema({
+            id: "mock123",
+            label: "Email address",
+            uiType: EElementType.EMAIL,
+            validation: [
+                {
+                    required: true,
+                    errorMessage: "Email address is required",
+                },
+            ],
+        }),
+    }
+);
+
+const MOCK_EMAIL_SCHEMA_WITH_PREVIOUS_SOURCE_PREFILL = generateMockSchema(
+    {
+        mock123: [
+            {
+                prefillMode: "Previous source",
+                actionId: "testaction",
+                path: "testpath",
+            },
+        ],
+    },
+    {
+        ...generateMockElementSchema({
+            id: "mock123",
+            label: "Email address",
+            uiType: EElementType.EMAIL,
+            validation: [
+                {
+                    required: true,
+                    errorMessage: "Email address is required",
+                },
+            ],
+        }),
+    }
+);
```

Comment:
since all the mock elements and schemas are used only once, you can declare them within the test case using them. we want to declare outside only if they are shared among multiple test cases